scripts/qiita

```diff
@@ -319,18 +319,18 @@ def start(port, master):
             if 'error_msg' in msg:
                 job_error = msg['error_msg']
 
-            qid = qdb.processing_job.ProcessingJob.by_ext_id(msg['Job_Id'])
-
-            if qid:
-                # Qiita recognizes the external ID from Watcher, and has
-                # mapped it to a Qiita ID. Now, create a Job object to
-                # register the job's changed status in Qiita.
-                job_obj = qdb.processing_job.ProcessingJob(qid)
-                job_obj.complete(job_state, error=job_error)
-            else:
+            try:
+                qjob = qdb.processing_job.ProcessingJob.by_external_id(
+                    msg['Job_Id'])
+            except Exception:
```

Comment:
What exceptions are expected here? From the code in `by_external_id` it seems like there aren't any exceptions raised, right?

Reply:
Excellent question! Note that I was trying to be overly cautions. Anyway, the original `by_external_id` returned the job id and if it didn't existed it returned empty string; then inside the `if` it will instantiate the job. Thus, I thought it will be better to instantiate the job inside the new `by_external_id`, which can raise an issue if the job-id doesn't exist - this is actually yielded by the class instantiation. Should I limit to just `QiitaDBUnknownIDError`?

Reply:
Thanks for the explanation, I think limiting to `QiitaDBUnknownIDError` would be best.